documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc

```diff
@@ -346,13 +347,30 @@ public class CarMapperImpl implements CarMapper {
     public PersonDto personToPersonDto(Person person) {
         //...
     }
+
+    private EngineDto engineToEngineDto(Engine engine) {
+        if ( engine == null ) {
+            return null;
+        }
+
+        EngineDto engineDto = new EngineDto();
+
+        engineDto.setHorsePower(engine.getHorsePower());
+        engineDto.setFuel(engine.getFuel());
+
+        return engineDto;
+    }
 }
 ----
 ====
 
 The general philosophy of MapStruct is to generate code which looks as much as possible as if you had written it yourself from hand. In particular this means that the values are copied from source to target by plain getter/setter invocations instead of reflection or similar.
 
-As the example shows the generated code takes into account any name mappings specified via `@Mapping`. If the type of a mapped attribute is different in source and target entity, MapStruct will either apply an automatic conversion (as e.g. for the _price_ property, see also <<implicit-type-conversions>>) or optionally invoke another mapping method (as e.g. for the _driver_ property, see also <<mapping-object-references>>).
+As the example shows the generated code takes into account any name mappings specified via `@Mapping`.
+If the type of a mapped attribute is different in source and target entity,
+MapStruct will either apply an automatic conversion (as e.g. for the _price_ property, see also <<implicit-type-conversions>>)
+or optionally invoke / create another mapping method (as e.g. for the _driver_ / _engine_ property, see also <<mapping-object-references>>).
+MapStruct will only create a new mapping method if and only if the source and target property adhere to the bean convention.
```

Comment:
I am not sure what @sjaakd meant with that. Maybe we need to say: "MapStruct will only create a new mapping method if and only if the source and target properties match".  

I think you know what I mean by match (field matches a getter for it, custom naming strategy will make 2 properties match etc.). Do you have some other suggestion for this? How should we write it?